### PR TITLE
docs: add Memfault OTA documentation

### PR DESCRIPTION
**Pull request by gminn**
- Sort the list of OTA providers alphabetically
- Add Memfault OTA blurb

### CODE REVIEW

File: doc/services/device_mgmt/ota.rst [hunk 1]
```diff
@@ -69,17 +79,24 @@ upgrade occurs.
...
 
 .. _MCUboot bootloader: https://mcuboot.com/
+.. _Eclipse hawkBit: https://www.eclipse.org/hawkbit/
 .. _Golioth: https://golioth.io/
 .. _Golioth Firmware SDK repository: https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/zephyr/fw_update
 .. _Golioth OTA documentation: https://docs.golioth.io/device-management/ota
-.. _Eclipse hawkBit: https://www.eclipse.org/hawkbit/
+.. _Memfault: https://memfault.com/
+.. _Memfault's OTA documentation: https://docs.memfault.com/docs/platform/ota
+.. _OTA client: https://github.com/memfault/memfault-firmware-sdk/tree/master/ports/zephyr/include/memfault/ports/zephyr/fota.h
```
Comment (noahp) [inline doc/services/device_mgmt/ota.rst:100]:
```suggestion
.. _Memfault OTA client: https://github.com/memfault/memfault-firmware-sdk/tree/master/ports/zephyr/include/memfault/ports/zephyr/fota.h
```

File: doc/services/device_mgmt/ota.rst [hunk 2]
```diff
@@ -36,27 +47,26 @@ upgrade occurs.
...
 
-`UpdateHub`_ is a platform for remotely updating embedded devices. Updates can
-be manually triggered or monitored via polling. When a new update is detected,
-the binary is downloaded and installed. MCUboot can be used to verify the
-signature before upgrading the firmware.
-
-There is an :zephyr:code-sample:`updatehub-fota` sample included in the Zephyr
-:ref:`mgmt-samples` section.
+Zephyr projects that use MCUboot and have a direct internet connection can leverage Memfault's
+built-in `OTA client`_ to download a payload from Memfault's OTA service, download it into
```
Comment (noahp) [inline doc/services/device_mgmt/ota.rst:66]:
```suggestion
built-in `OTA client`_ to download a payload from Memfault's OTA service, load it into
```

**Review by noahp — APPROVED** [review]

File: doc/services/device_mgmt/ota.rst [hunk 3]
```diff
@@ -36,27 +47,26 @@ upgrade occurs.
...
-the binary is downloaded and installed. MCUboot can be used to verify the
-signature before upgrading the firmware.
-
-There is an :zephyr:code-sample:`updatehub-fota` sample included in the Zephyr
-:ref:`mgmt-samples` section.
+Zephyr projects that use MCUboot and have a direct internet connection can leverage Memfault's
+built-in `Memfault OTA client`_ to download a payload from Memfault's OTA service, load it into
+the secondary partition, and then reboot into the new image. A working sample that
+leverages this client is available in the `Memfault Zephyr ESP32 sample`_ repository.
+`Memfault's OTA documentation`_ provides more information on how to deploy and manage releases.
```
Comment (nordicjm) [inline doc/services/device_mgmt/ota.rst:69]: is there any reason this can't be moved to the zephyr repo?

Comment (gminn) [reply doc/services/device_mgmt/ota.rst:69]: That is a great idea! What is the process for adding a sample app to Zephyr?

Comment (gminn) [reply doc/services/device_mgmt/ota.rst:69]: I would love to get this app into zephyr but I may need some guidance as it relies on a currently external module (Memfault sdk). If we add this sample to the zephyr app, perhaps we would also want to add Memfault to zephyr's `west.yml`? Unsure what the process is there so I'm very open to your suggestions & guidance!

Comment (nordicjm) [reply doc/services/device_mgmt/ota.rst:69]: Would have to read and follow this: https://docs.zephyrproject.org/latest/contribute/external.html would need to submit an issue with the details of the module for discussion with TSC

Comment (gminn) [reply doc/services/device_mgmt/ota.rst:69]: Great, thank you! I can certainly do that. Do you consider that a prerequisite of this PR or is it fine to add the documents with the out-of-tree sample app, and update them when it is added?

Comment (nordicjm) [reply doc/services/device_mgmt/ota.rst:69]: I would rather not personally, we have samples in tree and this should be there and available for all platforms. Having an external repo whereby it works for one board/SoC? That's not really useful in my mind

**Comment by nandojve** [comment]
I started to see Company names inside Zephyr general documentation. I think this is a bad move for Open Source and we should reconsider this kind of thing. We should even remove existent references to companies and all links to externals if they exists. I believe everything that is not in tree should not be added.

For me, this imply ethical and moral questions that passes through. At end, Zephyr is not a Portal or a Open Store to promote companies.

For instance, Update Hub was developed by OS Systems and besides that you can not find references to the company, only to the software name.

**Comment by gminn** [comment]
Closing since a minimum requirement for this PR would be an example app. There may also be a concern about using company names in Zephyr docs as raised by @nandojve.

**Pull request closed without merging**